### DISCUSSION TITLE
👍询问一下该项目Api设计的RoadMap

### DISCUSSION THREAD

**Discussion by rust-kotlin** [ThriceCola/Kovi#2] [General]
目前看到的一个比较新的Rust Onebot Sdk，一直苦于没有合适的rust库，期待该项目未来的发展。

比如：
- 在模式匹配上下功夫，添加更多的预置函数和链式调用，简化代码的书写
- 更多地使用回调函数和匿名函数的方法
- 类似router的设计来确定handler调用顺序，也可以参考axum的extractor设计

这只是我的一点想法，有时间再去细看下你的项目代码。
我曾用Go的Zerobot开发，整体开发体验还是不错的，你可以看下api调用部分的[代码](https://github.com/rust-kotlin/gpt-bot/blob/master/connect/connect.go)

**Comment by ThriceCola** [comment]
我有看到你fork我的项目，可惜早看了一个小时，在刚更新的0.1.0版本已经添加了更多的api。这些api都是无需关照返回值的api，下一版本会是将所有有用的有返回值的 OneBot api 封装进来。再接着才是更多预制函数和链式调用。

这个项目刚刚开发7天，还是非常简陋的形态。但是目前的插件开发体验跟未来会是一致的。

**Comment by rust-kotlin** [comment]
我看了下你的仓库代码，有兴趣参与开发。可以问下可以用tls加密的wss协议吗？我在测试过程中想直接连接我远程的服务器，想要加密一下数据传输。我看了下Onebot协议好像没有提https或者wss，可能具体的实现也不会支持加密的协议。

另外，clippy自动产生了部分代码提示，大概意思是：
- self.name == None 可以用 self.name.is_none()函数
- 在函数参数中我们一般不用&String类型，一般用&str，因为在外部调用时如果是&String会自动类型转换为&str
- 函数尾部的返回值一直不用写return Ok(()); ，直接写Ok(())就行了

**Comment by ThriceCola** [comment]
这些都是一些小问题，一小会就可以解决的。如果想对此项目做出贡献的话，请等待有需要解决的issus。
如果你有更好的想法实现，可以在discussions讨论。

另外OneBot要求的是http、ws实现，部分OneBot可能会有wss实现，当为了兼容性，暂时需要考虑是否要wss。

**Comment by ThriceCola** [comment]
这里是issues，讨论的话请转向discussions。

**Comment by ThriceCola** [comment]
已经转换成discussions了，能否加一下Q群857054777，我们可以及时的详细聊聊
